recipe-client-addon/lib/AddonStudies.jsm

```diff
@@ -112,6 +124,46 @@ this.AddonStudies = {
     };
   },
 
+  async init() {
+    // If the add-on an active study has been removed since we last ran, stop
+    // the study.
+    const activeStudies = (await this.getAll()).filter(study => study.active);
+    const db = await getDatabase();
+    for (const study of activeStudies) {
+      const addon = await AddonManager.getAddonByID(study.addonId);
+      if (!addon) {
+        this.markAsEnded(study);
+        await getStore(db).put(study);
+        Services.obs.notifyObservers(study, STUDY_ENDED_TOPIC);
```

Comment:
Can we just do this in `markAsEnded` directly? Updating the study based on an optional (default to this.db) passed DB connection would also reduce duplication and reduce the risk that we later add some other callsite for markAsEnded that doesn't do this.